fcp-common-pipeline.yaml

```
@@ -18,80 +21,80 @@ resources:
-        displayName: "Prepare and Init Workspace"
         steps:
+          - checkout: self
+          - checkout: FcpPipelineCommon
+
           - script: |
-              echo -e "\033[0;32mRepository: $(Build.Repository.Name)\033[0m"
+              echo APP_NAME=$(appName)
+              ls -la
+              env | grep -i "BUILD_REPOSITORY_NAME"
               echo -e "\033[0;33mBranch: $(Build.SourceBranchName)\033[0m"
-            displayName: "Print repo and branch"
+              echo ""
+              env
+            displayName: 'Build and Workspace Info'
 
+          - template: /ado-pipeline/init/app-version.yaml
+            parameters:
+              appFrameworkType: ${{ parameters.appFrameworkType }}
 
-  - stage: Build
-    dependsOn:
-      - Init
-    displayName: "Build Stage"
-    variables:
-      - template: variables.yml
-    jobs:
-      - job: BuildingApplication
-        displayName: "Build"
-        steps:
-          - checkout: self
-          - checkout: FcpPipelineCommon
           - template: /ado-pipeline/build/build-${{ parameters.appFrameworkType }}.yaml
             parameters:
               npmLegacyPeerDeps: ${{ parameters.npmLegacyPeerDeps }}
+          - ${{ if ne(variables['isPackage'], true) }}:
+              - template: /ado-pipeline/build/docker-build.yaml
+                parameters:
+                  dockerFilePath: ${{ parameters.dockerFilePath }}
 
-  - stage: stage
-    dependsOn:
-      - Init
-    displayName: "Stage"
+  - stage: Check
+    displayName: 'Check'
+    dependsOn: Build
     variables:
-      - template: variables.yml
+      - template: variables.yaml
     jobs:
       - job: SnyckScanApplication
-        displayName: "Snyk Application Scan"
+        displayName: Snyk Application Scan
         steps:
           - checkout: self
           - checkout: FcpPipelineCommon
           - template: /ado-pipeline/stage/snyk-scan-application.yaml
             parameters:
               appFrameworkType: ${{ parameters.appFrameworkType }}
       - job: SonarCloud
-        displayName: "SonarCloud Analysis"
+        displayName: SonarCloud Analysis
         steps:
           - checkout: self
           - checkout: FcpPipelineCommon
           - template: /ado-pipeline/stage/sonar-cloud.yaml
             parameters:
               appFrameworkType: ${{ parameters.appFrameworkType }}
 
-  - stage: Deployment
-    dependsOn:
-      - Init
-      - Build
-      - stage
-    displayName: "Deployment Stage"
-    variables:
-      - template: variables.yml
-    jobs:
-      - job: Deploy
-        displayName: "Deploy"
-        condition: succeeded()
-        steps:
-          - checkout: self
-          - checkout: FcpPipelineCommon
-          - template: /ado-pipeline/deploy/deploy-${{ parameters.type }}.yaml
\ No newline at end of file
+  - ${{ if eq(variables['isPackage'], true) }}:
+      - stage: PackageDeployment
+        condition: and( succeeded() , ne(variables['isPackage'], true) )
```

```suggestion
        condition: succeeded()
```
already a condition in the if statement above.